### PR TITLE
Add arm64 support for kfserving

### PR DESCRIPTION
**Pull request by MrXinWang**
Signed-off-by: Henry Wang <<henry.wang@arm.com>>

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/507)

### CODE REVIEW

**Comment by MrXinWang** [comment]
Hi @jinchihe! I think [something strange](https://github.com/kubeflow/tf-operator/pull/1098#issuecomment-548701876) is going on with the CI? Any ideas?

**Comment by jinchihe** [comment]
@MrXinWang That's should be caused by CI test env problem, I'm going to take a look.

**Comment by MrXinWang** [comment]
@jinchihe Thanks very much!

**Comment by rakelkar** [comment]
Cool what's the scenario? Do istio and knative already work on arm? Also a lot more will be required including possibily selecting different base images.. can you raise an issue with details on the use case?

/hold

**Comment by MrXinWang** [comment]
Hi @rakelkar! I think [issue#2237](https://github.com/kubeflow/kubeflow/issues/2337#issuecomment-524690065) and [this comment ](https://github.com/kubeflow/kubeflow/issues/2337#issuecomment-524690065) show something that you may be interested in.

Basically we are trying to enable kubeflow on arm64 platform, and yes you are absolutely right that a lot more work is required. When I started to work on this project the latest version of kubeflow was 0.5.1, and istio was introduced in 0.6.0 and later. For now I am just contributing what I have achieved to upstream. We do have some discussions about the base image in kubeflow/pipeline [PR#2507](https://github.com/kubeflow/pipelines/pull/2507), maybe you can also share your opinion about it.

**Comment by MrXinWang** [comment]
@jinchihe I think the test should be ok now.

/retest

**Comment by ellis-bigelow** [comment]
Test flake resolved in another PR. You may want to rebase.
/retest

File: Dockerfile [hunk 1]
```diff
@@ -9,6 +9,8 @@ COPY vendor/ vendor/
 # Build
 RUN if [ "$(uname -m)" = "ppc64le" ]; then \
         CGO_ENABLED=0 GOOS=linux GOARCH=ppc64le go build -a -o manager ./cmd/manager; \
+    elif [ "$(uname -m)" = "aarch64" ]; then \
+        CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -a -o manager ./cmd/manager; \
```
Comment (rakelkar) [inline Dockerfile]: make sense to just make this an envvar that gets picked up?

**Comment by rakelkar** [comment]
Cool okay - will be awesome to see it work on arm64. If you are trying to minimize initial scope you dont need the executor as it is not currently used.

/hold cancel

File: Dockerfile (hunk 1 above)
Comment (MrXinWang) [reply Dockerfile]: Yes, it definately makes sense, however referencing [moby/issue#29110](https://github.com/moby/moby/issues/29110#issuecomment-453718228) I don't think docker currently support the function that set ENV as the result of the shell command (in our case `uname -m`). I think this is also the reason why PowerPC guys keep the coding style of branch as`if ["$(uname -m)" = ARCH]` stuff.

Comment (MrXinWang) [reply Dockerfile]: if we use `RUN export MACHINE_ARCH=$(uname -m) && if [$MACHINE_ARCH]=ppc64le/aarch64/x86_64]; then blablabla`, I think it is a little bit redundant isn't it?

**Comment by MrXinWang** [comment]
@jinchihe Test works perfectly now, thanks

**Pull request merged**